src/tak/Client.java

```diff
@@ -488,6 +496,35 @@ else if (game!=null && (m = resignPattern.matcher(temp)).find() && game.no == In
                             other.game = null;
                         }
                     }
+                    //Mark/Unmark target board field.
+                    else if (game != null && (m=markPattern.matcher(temp)).find() && game.no == Integer.parseInt(m.group(1))) {
+                      String function = m.group(2);
+                      String field = m.group(3).toLowerCase();
+                      if (game.setMarked(function.equals("Mark"), field, player))
+                            sendOK();
+                      else
+                            sendNOK();
+                    }
+                    //Relay ingame chat.
+                    else if (game != null && (m=ingamePattern.matcher(temp)).find() && game.no == Integer.parseInt(m.group(1))) {
+                        String msg = "<" + player.getName() + "> " + m.group(2);
+                        if (game.white == player || game.black == player) {
+                            msg = " PlayerChat " + msg;
+                            if(!player.isGagged()) {
+                                game.sendToOtherPlayer(game.white, "Game#" + game.no + msg);
```

Comment:
The ID of the game is not per se required to display a message, but it might come the time, when different games have different ingame chat windows. For that eventuality, the ID is transmitted.